File: db/event_person.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_person.
*/

class Event_Person extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "event_person";
    $this->order = "";
    $this->domain = "event";
    $this->field['event_person_id']['type'] = 'SERIAL';
    $this->field['event_person_id']['not_null'] = true;
    $this->field['event_id']['type'] = 'INTEGER';
    $this->field['event_id']['not_null'] = true;
    $this->field['person_id']['type'] = 'INTEGER';
    $this->field['person_id']['not_null'] = true;
    $this->field['event_role_id']['type'] = 'INTEGER';
    $this->field['event_role_id']['not_null'] = true;
    $this->field['event_role_state_id']['type'] = 'INTEGER';
    $this->field['remark']['type'] = 'TEXT';
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['event_person_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/event_person.rb
module Momomoto
  class Event_person < Base
    def initialize
      super
      @domain = 'event'
      @log_changes = true
      @fields = {
        :event_person_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :event_id => Datatype::Integer.new( {:not_null=>true} ),
        :person_id => Datatype::Integer.new( {:not_null=>true} ),
        :event_role_id => Datatype::Integer.new( {:not_null=>true} ),
        :event_role_state_id => Datatype::Integer.new( {} ),
        :remark => Datatype::Text.new( {} ),
        :rank => Datatype::Integer.new( {} ),
        :last_modified => Datatype::Timestamp.new( {:with_timezone=>true, :not_null=>true, :default=>true, :auto_update=>true} ),
        :last_modified_by => Datatype::Integer.new( {} )
      }
    end
  end
end
